src/threads/decimal.h

```diff
@@ -0,0 +1,21 @@
+typedef struct
```

Comment:
I think we could just use:
```
typedef int decimal
```
(by default integer is signed)

With this approach these methods are not needed:
```
decimal div_int(decimal a, int b);
decimal mult_int(decimal a, int b);
decimal subtract_int(decimal a, int b);
decimal sum_int(decimal a, int b);
```
because these operations should just work (please check)

Reply:
I used `struct` instead of `typedef` to add static types to reduce amount of errors.

`decimal` here is not the same as `int` and any operation with `decimal` must be done using methods from `decimal.c` and decimal must be passed only to parameters with `decimal` type. When `typedef int decimal` is used it is too easy to pass `int` instead of `decimal` (and it should be prohibited)

For example for int `1` decimal will be `16384`

Reply:
I was not sure that it has potential performance implications, but look on this experiment :)

https://godbolt.org/z/hj817TzxT

Look at the assembler. It does not consider field from the same structure as the same and has to put the same value to a different register before multiplication ))

(I understand it does not much matter for our case, just to satisfy engineering curiosity)

Reply:
And one more interesting observation ))
With `-O1` compiler optimization flag these functions become identical: https://godbolt.org/z/Ef5YKe7qh

```
f_1:
        imul    edi, edi
        mov     eax, edi
        ret
f_2:
        imul    edi, edi
        mov     eax, edi
        ret
```